app/models/search_builder.rb

```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 class SearchBuilder < Blacklight::SearchBuilder
   include Blacklight::Solr::SearchBuilderBehavior
+  include Hydra::AccessControlsEnforcement
+
+  def add_access_controls_to_solr_params(*args)
+    return if current_ability.can?(:manage, Valkyrie::Model)
+    apply_gated_discovery(*args)
+  end
```

Comment:
This is a pass-by-default scheme with a public method here. Granted, overriding this would be a mouthful, but someone would have to be certain to `apply_gated_discovery` properly or risk things being exposed accidentally. Is a reject-by-default scheme more appropriate?